File: 7/7.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(long long n) {
	if (n == 2) return true;
	for (int i=2; i <= ceil(sqrt(n)); i++)
		if (n%i == 0) return false;
	return true;	
}

long long primes(int n) {
	if (n == 1) return 2;
	
	int count=1, i=3;
	while (true) {
		if (is_prime(i)) count++;
		if (count == n) return i;

		i += 2;
	}
}

int main() {
	int n = 10001;
	cout << primes(n) << endl; 
	return 0;
}

File: 7/7.py
from math import ceil, sqrt
import itertools

def is_prime(n: int) -> bool:
	if n == 2: return True

	for i in range(2, ceil(sqrt(n))+1):
		if n % i == 0: return False

	return True


def primes(n: int) -> int:
	if n == 1: return 2

	count: int = 1
	for i in itertools.count(3, 2):
		if is_prime(i): count += 1
		if count == n: return i


n: int = 10001
print(primes(n))
